broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/steps/StoragePartitionTransitionStep.java

```diff
@@ -1,59 +0,0 @@
-/*
- * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
- * one or more contributor license agreements. See the NOTICE file distributed
- * with this work for additional information regarding copyright ownership.
- * Licensed under the Zeebe Community License 1.1. You may not use this file
- * except in compliance with the Zeebe Community License 1.1.
- */
-package io.camunda.zeebe.broker.system.partitions.impl.steps;
-
-import io.atomix.raft.RaftServer.Role;
-import io.camunda.zeebe.broker.system.partitions.PartitionTransitionContext;
-import io.camunda.zeebe.broker.system.partitions.PartitionTransitionStep;
-import io.camunda.zeebe.util.sched.future.ActorFuture;
-import io.camunda.zeebe.util.sched.future.CompletableActorFuture;
-
-public class StoragePartitionTransitionStep implements PartitionTransitionStep {
-
-  @Override
-  public ActorFuture<Void> transitionTo(
-      final PartitionTransitionContext context, final long term, final Role role) {
-
-    switch (role) {
```

Comment:
omg what was this :see_no_evil: :D

Reply:
This is the future! :slightly_smiling_face: 
@pihme has introduced a new `PartitionTransitionStep` in which we can react to role transitions. So instead of closing and restarting services, we will be able to just transition to a new mode. The one that is deleted now is related to snapshot replication. It was using this transition behavior. I think that we  should move other partitions steps also to `PartitionTransitionStep` in the follow up issue to extract bootstrap and transition steps.